tests/periph_wdg/tests/01-run.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2019 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import sys
+import json
+from testrunner import run
+
+# We test only up to 10ms, with smaller times mcu doesn't have time to
+# print sysyem time before resetting
+reset_times_us = [1e4, 5e4, 1e5, 5e5, 1e6]
+
+
+def get_reset_time(child):
+    reset_time = 0
+    reset = False
+    while not reset:
+        line = child.readline()
+        if "main(): This is RIOT!" in line:
+            break
+        jason = line.strip().split("json:")
+        reset_time = json.loads(jason[1])
+    return reset_time['time']
+
+
+def get_start_time(child):
+    child.readline()
+    child.readline()
+    line = child.readline().strip().split("json:")
+    start_time = json.loads(line[1])
+    return start_time['starttime']
+
+
+def testfunc(child):
+    for rst_time in reset_times_us:
+        child.sendline("setup 0 {}".format(rst_time))
+        child.expect_exact("[wdg]: - ")
+        child.expect_exact(" -")
+
+        if child.before != "invalid configuration time":
```

Comment:
You don't need this check. This (minor) case can simply be tested sending and invalid configuration using `sendline` and test the output is as expect using `expect_exact` or a regexp in `expect`.

Reply:
Invalid configuration time will be different depending on each individual wdg.

I could move IWDG_MIN_TIME_US and IWDG_MAX_TIME_US to periph_cpu_common.h and therefore define invalid ranges from capturing so define values on start up. What do you think?

Reply:
I think it's a good idea :) These values could be asserted as well when configuring the watchdog.